lib/db.js

```diff
@@ -349,44 +367,32 @@ class LimitDBRedis extends EventEmitter {
     }
 
     const count = params.count || 50;
-    const transform = this.buildStatusResponse(bucket, prefixLength);
+    const transform = this.buildStatusResponse(params.type, bucket, prefixLength);
 
-    const nodes = this.redis.nodes ? this.redis.nodes('slave') : [this.redis];
-    const errors = [];
-    async.map(nodes, (node, cb) => {
-      if (!node.bucketStatus) {
-        node.defineCommand('bucketStatus', {
-          numberOfKeys: 0,
-          lua: STATUS_LUA
-        });
-      }
-      node.bucketStatus(prefix, count, (err, results) => {
-        if (err) {
-          errors.push(err);
-          this.emit('error', err);
-        }
-        return cb(null, results || []);
-      });
-    }, (err, results) => {
-      if (err) {
-        // Should not happen
-        return callback(err);
-      }
+    const matches = [];
+    let errors = [];
+    const stream = this.scanStream({ prefix });
 
-      const items = [];
-      for (let i = 0; i < results.length; i++) {
-        for (let j = 0; j < results[i].length; j++) {
-          const match = results[i][j];
-          items.push(transform(match[2], parseInt(match[0], 10), parseInt(match[1], 10)));
-          if (items.length === count) {
-            break;
-          }
-        }
-        if (items.length === count) {
+    stream.on('error', (err) => {
+      errors.push(err);
+    });
+
+    stream.on('data', (keys) => {
+      for (let i = 0; i < keys.length; i++) {
+        matches.push(keys[i]);
+        if (matches.length >= count) {
           break;
         }
       }
-      return callback(null, { items, errors });
+      if (matches.length >= count) {
+        stream.end();
```

Comment:
I might be wrong, but I don't think this is fine, the writer side is the one in charge of ending the stream, if the reader side `ends` it the result is undefined I think; you might even end up with a `write after end` error I think.

The reader side should simply stop consuming the stream (`stream.pause()`) that way the writter will eventually stop requesting data and eventually the stream will be GC'd

Reply:
Yep, that error would get bubbled up to the on error listener.

Looking for a way to exit quicker. Pause will simply stop calling the data listener but will actually collect data internally thus wasting CPU, memory and Redis calls (particularly this one :D).

Reply:
I believe since its a Passthrough stream I could use `destroy`.